src/main/java/store/reduct/client/config/ServerProperties.java

```diff
@@ -1,11 +1,51 @@
 package store.reduct.client.config;
 
-public record ServerProperties(boolean isSsl, String host, int port) {
+import java.util.Objects;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
+import store.reduct.common.exception.ReductException;
+import store.reduct.utils.Strings;
+
+@AllArgsConstructor
+@Builder
+public class ServerProperties {
+	private Boolean isSsl;
+	private String host;
+	private Integer port;
+	private String url;
+	@Getter
+	private String apiToken;
 
 	/**
+	 * Returns a base url for the server. Initialize isSsl, host, port fields before
+	 * use the method.
+	 * 
 	 * @return The base url for the server
 	 */
 	public String getBaseUrl() {
```

Comment:
we don't need to split the URL into components and then rebuild. Just create two fields: `url` and `apiToken`.  For users, it is also better:

```java
ServerProperties.url("https://play.reduct.store").apiToken("reductstore").
```
looks better than
```java
ServerProperties.host("play.reduct.store").port(443).isSsl(true).apiToken("reductstore").

```

Reply:
done